docs/references/contribution_guide.md

```diff
@@ -0,0 +1,165 @@
+
+> **Note**: SGLang does **not** accept PRs on the main repo. Please fork the repository under your GitHub account, then clone your fork locally.
+
+```bash
+git clone https://github.com/<your_user_name>/sglang.git
+cd sglang
+```
+
+### 1.2 Install Dependencies & Build
+
+Refer to our [Install SGLang](https://sgl-project.github.io/start/install.html) documentation for more details on setting up the necessary dependencies.
+
+Before installing packages like FlashInfer, ensure your local PyTorch and CUDA versions match:
+
+```bash
+python -c "import torch; print('PyTorch Version:', torch.__version__); print('CUDA Version:', torch.version.cuda)"
+nvidia-smi
+```
+
+Based on this information, choose the correct FlashInfer package (or any other CUDA-dependent library).
+Below is an example assuming you have **PyTorch 2.4** with **CUDA 12.4**:
+
+```bash
+# Example: Installing FlashInfer (Pick the right version for your CUDA/PyTorch)
+pip install --upgrade pip
+pip install -e "python[all]" --find-links https://flashinfer.ai/whl/cu124/torch2.4/flashinfer/
+
+# Now install the latest SGLang
+cd sglang/python
+pip install .
+```
+---
+
+## 2. Code Formatting with Pre-Commit
+
+We use [pre-commit](https://pre-commit.com/) to maintain consistent code style checks. Before pushing your changes, please run:
+
+```bash
+pip3 install pre-commit
+cd sglang
+pre-commit run --all-files
+```
+
+- **`pre-commit run --all-files`** manually runs all configured checks, applying fixes if possible.
+- If it fails the first time, re-run it to ensure lint errors are fully resolved.
+- Make sure your code passes all checks **before** creating a Pull Request.
+- **Do not commit** directly to the `main` branch. Always create a new branch (e.g., `feature/my-new-feature`), push your changes, and open a PR from that branch.
+
+---
+
+## 3. Writing Documentation & Running Docs CI
+
+Most documentation files are located under the `docs/` folder. We prefer **Jupyter Notebooks** over Markdown so that all examples can be executed and validated by our docs CI pipeline.
+
+### 3.1 Docs Workflow
+
+```bash
+# 1) Compile all Jupyter notebooks
+make compile
+
+# 2) Generate static HTML
+make html
+
+# 3) Preview documentation locally
+bash serve.sh
+# Open your browser at the displayed port to view the docs
+
+# 4) Clean notebook outputs
+pip install nbstripout
+find . -name '*.ipynb' -exec nbstripout {} \;
+# nbstripout removes notebook outputs so your PR stays clean
+
+# 5) Pre-commit checks and create a PR
+pre-commit run --all-files
+# After these checks pass, push your changes and open a PR on your branch
+```
+
+- Add or update your Jupyter notebooks in the appropriate subdirectories under `docs/`.
+- If you add new files, remember to update `index.rst` (or relevant `.rst` files) accordingly.
+
+- For tasks that require a running SGLang backend, you might launch a server process or an engine instance. Once you finish testing or using it, **remember to shut it down**. Refer to:
```

Comment:
If you need to run and shut up a SGLang server or engine, following these examples:

1. Launch and close Sever:

```bash
#Launch Sever

from sglang.utils import (
    execute_shell_command,
    wait_for_server,
    terminate_process,
    print_highlight,
)

server_process = execute_shell_command(
    "python -m sglang.launch_server --model-path meta-llama/Meta-Llama-3.1-8B-Instruct --port 30000 --host 0.0.0.0"
)

wait_for_server("http://localhost:30000")

# Terminate Sever

terminate_process(server_process)
```

2. Launch Engine

```bash
# Launch Engine

import sglang as sgl
import asyncio

llm = sgl.Engine(model_path="meta-llama/Meta-Llama-3.1-8B-Instruct")

# Terminalte Engine
llm.shutdown()
```